subsys/bluetooth/mesh/settings.c

```diff
@@ -142,8 +143,8 @@ static void store_pending(struct k_work *work)
 {
 	BT_DBG("");
 
-	if (atomic_test_and_clear_bit(pending_flags,
-				      BT_MESH_SETTINGS_RPL_PENDING)) {
+	if (IS_ENABLED(CONFIG_BT_MESH_RPL_STORAGE_MODE_SETTINGS) &&
+	    atomic_test_and_clear_bit(pending_flags, BT_MESH_SETTINGS_RPL_PENDING)) {
 		bt_mesh_rpl_pending_store(BT_MESH_ADDR_ALL_NODES);
```

Comment:
Should there be a IS_ENABLED around the declaration of "bt_mesh_rpl_pending_store" in bluetooth/mesh/main.h?

Reply:
Not 100% sure, but I guess that it is not needed for the declaration